docs/stdlib/numbers.rst

```diff
@@ -814,41 +814,104 @@ from :eql:type:`str` and :eql:type:`json`.
 
 
 .. eql:function:: std::to_int16(s: str, fmt: optional str={}) -> int16
+                  std::to_int16(val: bytes) -> int16
 
     :index: parse int16
 
-    Returns an :eql:type:`int16` value parsed from the given string.
+    Returns an :eql:type:`int16` value parsed from the given input.
+
+    The string parsing function will use an optional format string passed as
+    *fmt*. See the :ref:`number formatting options
+    <ref_std_converters_number_fmt>` for help writing a format string.
+
+    .. code-block:: edgeql-repl
+
+        db> select to_int16('23');
+        {23}
+        db> select to_int16('23%', '99%');
+        {23}
+
+    The bytes conversion function expects big-endian representation. Only the
+    lower 2 bytes will be used to determine the value, the rest will be
+    ignored. This means that a 64-bit integer can be passed as long as the
+    actual value is contained in the lower 2 bytes.
+
+    .. code-block:: edgeql-repl
+
+        db> select to_int16(b'\x00\x07');
+        {7}
+        db> select to_int16(to_bytes(<int64>123));
```

Comment:
Hm, I'm not sure that's the behavior we want.  I would expect an error, not silent truncation.